lib/iris/cube.py

```diff
@@ -1639,7 +1639,7 @@ def dtype(self, dtype):
                 if not self.has_lazy_data():
                     emsg = 'Cube does not have lazy data, cannot set dtype.'
                     raise ValueError(emsg)
-                if dtype.kind != 'i':
+                if dtype.kind not in 'biu':
                     emsg = ('Can only cast lazy data to integral dtype, '
```

Comment:
Yup, you're totally right. `numpy` in it's wisdom doesn't actually have `bool_` subclassed under `integer`, see [here](http://numpy.readthedocs.io/en/latest/reference/arrays.scalars.html#) ... I'll update the `emsg`. Good spot.